guvnor-m2repo-editor/guvnor-m2repo-editor-api/src/main/java/org/guvnor/m2repo/preferences/ArtifactRepositoryPreference.java

```diff
@@ -24,20 +24,68 @@
 @WorkbenchPreference(identifier = "ArtifactRepositoryPreference", bundleKey = "ArtifactRepositoryPreference.Label")
 public class ArtifactRepositoryPreference implements BasePreference<ArtifactRepositoryPreference> {
 
-    @Property(bundleKey = "ArtifactRepositoryPreference.DefaultM2RepoDir")
-    private String defaultM2RepoDir;
+    @Property(bundleKey = "ArtifactRepositoryPreference.GlobalM2RepoDir")
+    private String globalM2RepoDir;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.GlobalM2RepoDirEnabled")
+    private boolean globalM2RepoDirEnabled;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.WorkspaceM2RepoDir")
+    private String workspaceM2RepoDir;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.WorkspaceM2RepoDirEnabled")
+    private boolean workspaceM2RepoDirEnabled;
+
+    @Property(bundleKey = "ArtifactRepositoryPreference.DistributionManagementM2RepoDirEnabled")
+    private boolean distributionManagementM2RepoDirEnabled;
 
     @Override
     public ArtifactRepositoryPreference defaultValue(final ArtifactRepositoryPreference defaultValue) {
-        defaultValue.defaultM2RepoDir = "repositories/kie";
+        defaultValue.globalM2RepoDir = "repositories/kie/global";
+        defaultValue.workspaceM2RepoDir = "repositories/kie/workspaces";
+        defaultValue.globalM2RepoDirEnabled = true;
+        defaultValue.workspaceM2RepoDirEnabled = false;
+        defaultValue.distributionManagementM2RepoDirEnabled = true;
```

Comment:
What's the "Distribution Management M2 Repository directory"? A Project's ```pom.xml``` chain (from project and through all it's parents) can declare a ```<distributionManagement>``` section containing details of where the JAR should be deployed when built. This can presumably be different for each "workspace" (as different Users can have different effective poms); but I wonder what purpose this directory serves!? The configured distribution management section could point to a remote server over HTTP so a local directory seems irrelevant - unless I'm missing something obvious.